### ISSUE TITLE
How critical is the heap-based evaluation?

### ISSUE THREAD

**Issue by bradezard131** [ppengtang/oicr#14]
I'm trying to re-implement your model under detectron2 to build on for my own research (trying to have it using an up-to-date version of pytorch, this seemed like the easiest route). So far the VGG16-based model is peaking at ~30% accuracy, rather than the reported 41.2%. I have tried a couple different sets of weights exported from Caffe as well as the Pytorch pretrained weights and they all land at the same point (although Caffe weights are prone to turning to NaN and crashing). I have tried re-initialising the dilated convs and keeping the pretrained weights, and I have still been unable to reproduce the results.

I don't have the heap-based heuristics since they aren't in the paper, so I figured it was not essential. I am wondering if I might be mistaken?

**Comment by ppengtang** [comment]
Hi, sorry for the delayed reply. Maybe you can refer to our [PyTorch implementation](https://github.com/ppengtang/pcl.pytorch), although it is based on an older version of PyTorch. It is also important to notice that you should use RoI Pooling instead of RoI Align, because some guys told me that RoI Align will result in bad results for WSOD.

I'm not quite sure about your meaning of "heap-based heuristics", could you give more details?
